src/parser/toplevel.lisp

```diff
@@ -1619,91 +1619,61 @@ consume all attributes")))
-    (error 'parse-error
-           :err (coalton-error
-                 :span (cst:source (cst:second method-form))
-                 :file file
-                 :message "Malformed method definition"
-                 :primary-note "expected docstring"
-                 :notes
-                 (list
-                  (make-coalton-error-note
-                   :type :secondary
-                   :span (cst:source (cst:second form))
-                   :message "in this class definition")))))
-
-  (let (docstring)
-    (when (and (cst:atom (cst:second method-form))
-               (stringp (cst:raw (cst:second method-form))))
-      (setf docstring (cst:raw (cst:second method-form))))
-
-    (make-method-definition
-     :name (make-identifier-src
-            :name (node-variable-name (parse-variable (cst:first method-form) file))
-            :source (cst:source (cst:first method-form)))
-     :docstring docstring
-     :type (parse-qualified-type (if docstring
-                                     (cst:third method-form)
-                                     (cst:second method-form))
-                                 file)
-     :source (cst:source method-form))))
+  (macrolet ((fail (local-form note)
+               `(error 'parse-error
+                       :err (coalton-error
+                             :span (cst:source ,local-form)
+                             :file file
+                             :message "Malformed method definition"
+                             :primary-note ,note
+                             :notes (list (make-coalton-error-note
+                                           :type :secondary
+                                           :span (cst:source (cst:second form))
+                                           :message "in this class definition"))))))
+    (let (docstring name type)
+      (case (consp-length method-form)
+        ((nil 0 1)                      ; method-form is not consp, or is too short
+         (fail method-form "missing method type"))
+        (2                              ; good case 1: (<name> <type>)
+         (unless (and (cst:atom (cst:first method-form))
+                      (identifierp (cst:raw (cst:first method-form))))
+           (fail (cst:first method-form) "expected symbol"))
+         (setf name (parse-variable (cst:first method-form) file))
+         (setf type (parse-qualified-type (cst:second method-form) file)))
+        (3                              ; good case 2: (<name> "docstring" <type>)
+         (unless (and (cst:atom (cst:first method-form))
+                      (identifierp (cst:raw (cst:first method-form))))
+           (fail (cst:first method-form) "expected symbol"))
+         (setf name (parse-variable (cst:first method-form) file))
+         (unless (and (cst:atom (cst:second method-form))
+                      (stringp (cst:raw (cst:second method-form))))
+           (fail (cst:second method-form) "expected docstring"))
```

Comment:
I think this should be changed to `"unexpected trailing form"` on the `third` element, as discussed below.

Reply:
done, as according to the comment on define-class.4.error